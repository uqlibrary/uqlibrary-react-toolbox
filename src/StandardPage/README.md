# StandardPage

component for a pre-formatted page

## Props

- `title: PropTypes.string.isRequired`

## Usage

**./App.js**
```jsx
import React from 'react';
...

import {StandardPage} from 'uqlibrary-react-toolbox';
        
const App = () => (
  <div>
    <StandardPage 
      title='Contact us'>
      some text or react elements....
    </StandardPage>
  </div>
);

ReactDOM.render(
  <App />,
  document.getElementById('app')
);
```

