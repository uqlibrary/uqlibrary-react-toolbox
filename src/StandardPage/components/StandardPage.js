import React from 'react';
import PropTypes from 'prop-types';

export default function StandardPage({title, className, children}) {
    return (
        <div className={`layout-fill ${className}`}>
            <h1 className="title is-3">{title}</h1>
            <div className="layout-card">
                {children}
            </div>
        </div>
    );
}

StandardPage.propTypes = {
    title: PropTypes.string,
    className: PropTypes.string,
    children: PropTypes.any
};
