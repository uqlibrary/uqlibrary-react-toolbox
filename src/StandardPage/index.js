export {default as StandardPage} from './components/StandardPage';
