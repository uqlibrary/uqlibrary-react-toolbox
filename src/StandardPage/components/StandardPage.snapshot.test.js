jest.dontMock('./StandardPage');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import StandardPage from './StandardPage';

function setup(title) {
    const props = {title};
    return shallow(<StandardPage {...props} />);
}

describe('Snapshot tests for StandardPage component', () => {
    it('renders StandardPage with title', () => {
        const wrapper = setup('standard page title');
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});