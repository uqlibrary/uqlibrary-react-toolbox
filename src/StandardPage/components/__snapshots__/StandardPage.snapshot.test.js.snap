// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests for StandardPage component renders StandardPage with title 1`] = `
<div
  className="layout-fill undefined"
>
  <h1
    className="title is-3"
  >
    standard page title
  </h1>
  <div
    className="layout-card"
  />
</div>
`;
